### DISCUSSION TITLE
Is it expected that using SPM packages the whole project?

### DISCUSSION THREAD

**Discussion by mikescamell** [touchlab/KMMBridge#206] [Q&A]
Hello! 

First, thanks for creating this library. It was very quick for me to get up and running to get this working and integrated in a little Sample project 🙌 

I just had a question around sharing using SPM. I followed the docs and setup KMMBridge in the common (shared) module and expected that to be the only thing packaged up. But when adding this to our iOS project, I noticed it's packaging the whole project, Android App included.

Is this to be expected using SPM or have I made a mistake?

Thanks.

**Comment by kpgalligan** [comment]
Sorry for the delay :)

> But when adding this to our iOS project, I noticed it's packaging the whole project, Android App included.

If you mean you see the whole project under the package in Xcode, that's an SPM/Xcode thing. To SPM, a dependency is a git repo. In the SPM ecosystem, dependencies are distributed as source code and built locally. The whole repo is considered the dependency. For KMP, we pre-build the Kotlin code and publish the binary. However, the "dependency" is the repo it is published on, which in this case is the whole Android repo. So, SPM/Xcode downloads all of the source, but it doesn't do anything with it.

If you don't want SPM/Xcode to do that, you'd need to publish the Kotlin builds to a separate repo.
